synapse/storage/databases/main/event_federation.py

```diff
@@ -1207,6 +1207,15 @@ def _get_next_staged_event_for_room_txn(txn):
 
         return origin, event
 
+    async def get_all_staged_rooms(self) -> List[str]:
```

Comment:
given that our database methods aren't really scoped, I think this needs a better name:

```suggestion
    async def get_all_rooms_with_staged_incoming_events(self) -> List[str]:
```